### PR TITLE
Assignment2A and Assignment2B

### PR DESCRIPTION
**Pull request by meta-nilesh-patel**

### CODE REVIEW

File: SCF/Assignment2A/HexCalc.java [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+		for(int charIndex = operand.length()-1; charIndex >=0; charIndex--)
+		{
+			
+			//@hexChar store character from given index
+			char hexChar = operand.charAt(charIndex);
+			
+			//if  @hexChar contain A-F than taking respective decimal from HashMap @hexToDecTable
+			if((int)hexChar >= asciiOfCharA && (int)hexChar <= asciiOfCharF)
+			{
+
```
Comment (rash-tiwari) [inline SCF/Assignment2A/HexCalc.java]: Remove this Extra Line

File: SCF/Assignment2A/HexCalc.java [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+		long decimal = Long.parseLong(operand);
+		
+		while(decimal != 0)
+		{
+			reminder = (int) decimal % base;
+			decimal = decimal / base;
+			if(reminder>=10 && reminder<=15)
+			{
+				hexString.append(HexCalc.decToHexTable.get(reminder));
+			}else
```
Comment (rash-tiwari) [inline SCF/Assignment2A/HexCalc.java]: Please move ```else``` in next line.

File: SCF/Assignment2A/TestHexCalc.java [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+			    		   /**
+			    		    * 
+				    	    * Printing result of addition
+				    	    */
+			    		   System.out.println(firstOperand+" + " + secondOperand + " = " + addResult);
+			    	   }
+			    	   catch(IOException io)
+			    	   {
+			    		   System.out.println(io);			    	   
+			    	   break;
```
Comment (rash-tiwari) [inline SCF/Assignment2A/TestHexCalc.java]: ```break``` should be outside of ```catch block```

File: SCF/Assignment2A/ArithmeticLogicalOperation.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	public String addOperation(String firstOperand , String secondOperand) throws IOException;
+	
+	/**
+	 * 
+	 * @param firstOperand operand for subtraction
+	 * @param secondOperand operand for subtraction
+	 * @return result of subtraction of two operand
+	 * @throws IOException if parameter not contain legal value
+	 */
+	public String subtractOperation(String firstOperand , String secondOperand)throws IOException;
```
Comment (rash-tiwari) [inline SCF/Assignment2A/ArithmeticLogicalOperation.java]: Add Space before ```throws``` and correct it at other places as well.

File: SCF/Assignment2A/ArithmeticLogicalOperation.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	public String multiplicationOperation(String firstOperand , String secondOperand)throws IOException;
+	
+	/**
+	 * 
+	 * @param firstOperand operand for division
+	 * @param secondOperand operand for division
+	 * @return result of division of two operand
+	 * @throws IOException if parameter not contain legal value
+	 */
+	public String divisonOperation(String firstOperand , String secondOperand)throws IOException,ArithmeticException;
```
Comment (rash-tiwari) [inline SCF/Assignment2A/ArithmeticLogicalOperation.java]: Add space after each ```,``` and correct at other places as well.

File: SCF/Assignment2A/HexCalc.java [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+			return "0";
+		}
+		
+		int reminder;
+		StringBuilder hexString = new StringBuilder();
+		long decimal = Long.parseLong(operand);
+		
+		while(decimal != 0)
+		{
+			reminder = (int) decimal % base;
```
Comment (rash-tiwari) [inline SCF/Assignment2A/HexCalc.java:125]: Please correct spelling ```reminder ``` to ```remainder```

File: SCF/Assignment2A/HexCalc.java [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+		if(secondOperand.equals("0"))
+		{
+			throw new ArithmeticException("Divide by zero");
+		}
+		
+		long firstOperandDecimal = hexToDecimal(firstOperand);
+		long secondOperandDecimal = hexToDecimal(secondOperand);
+		long divResult = firstOperandDecimal / secondOperandDecimal;
+		return decimalToHex(Long.toString(divResult));
+
```
Comment (rash-tiwari) [inline SCF/Assignment2A/HexCalc.java:244]: remove this extra line.

File: SCF/Assignment2A/HexCalc.java [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+		secondOperand = removeInitialZeros(secondOperand);
+		//first checking if length is equal
+		if(firstOperand.length()<secondOperand.length())
+		{
+			return true;
+		}
+
+		if(firstOperand.length()==secondOperand.length())
+		{
+
```
Comment (rash-tiwari) [inline SCF/Assignment2A/HexCalc.java]: remove extra line from here and from similar places.

**Review by rash-tiwari — DISMISSED** [review]
Approved

**Comment by rash-tiwari** [comment]
Merged SFC Assignment 1 and Assignment 2

File: SCF/Assignment2A/HexCalc.java (hunk 1 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/HexCalc.java]: Done

File: SCF/Assignment2A/HexCalc.java (hunk 2 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/HexCalc.java]: Done

File: SCF/Assignment2A/TestHexCalc.java (hunk 3 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/TestHexCalc.java]: Done

File: SCF/Assignment2A/ArithmeticLogicalOperation.java (hunk 4 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/ArithmeticLogicalOperation.java]: Done

File: SCF/Assignment2A/HexCalc.java (hunk 6 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/HexCalc.java:125]: Done

File: SCF/Assignment2A/HexCalc.java (hunk 7 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/HexCalc.java:244]: Done

File: SCF/Assignment2A/HexCalc.java (hunk 8 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/HexCalc.java]: Done

File: SCF/Assignment2A/TestHexCalc.java [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+				    		   System.out.println(firstOperand+" is greater than "+secondOperand);
+				    	   }
+				    	   else
+				    	   {
+				    		   System.out.println(firstOperand+" is not greater than "+secondOperand);
+				    	   }
+			    	   }
+			    	   catch(IOException io)
+			    	   {
+			    		   System.out.println(io);
```
[the remark this answers is not in the document]
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/TestHexCalc.java]: Done

File: SCF/Assignment2A/ArithmeticLogicalOperation.java (hunk 5 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment2A/ArithmeticLogicalOperation.java]: Done

Comment (meta-nilesh-patel) [reply SCF/Assignment2A/ArithmeticLogicalOperation.java]: Done

**Review by rash-tiwari — APPROVED** [review]

**Pull request merged**